dataprofiler/profilers/profile_builder.py

```diff
@@ -1890,6 +1890,20 @@ def _get_unique_row_ratio(self) -> float:
             return len(self.hashed_row_dict) / self.total_samples
         return 0
 
+    def _get_sample_size(self, df_series: pd.Series) -> int:
+        """
+        Determine the minimum sampling size for detecting column type.
+
+        :param df_series: a column of data
+        :type df_series: pandas.core.series.Series
+        :return: integer sampling size
+        :rtype: int
+        """
+        len_df = len(df_series)
+        if len_df <= self._min_sample_size:
+            return int(len_df)
+        return max(int(self._sampling_ratio * len_df), self._min_sample_size)
+
```

Comment:
ideally we don't have this.... 🤔 about how to remove and just use the `BaseProfiler._get_sample_size`